File: WdtcCore/src/main/java/org/wdt/wdtc/core/download/infterface/InstallTaskInterface.java
package org.wdt.wdtc.core.download.infterface;

import java.io.IOException;

public interface InstallTaskInterface {
  void overwriteVersionJson() throws IOException;

  void writeVersionJsonPatches() throws IOException;

  void afterDownloadTask() throws IOException;

  void beforInstallTask() throws IOException;
}

File: WdtcCore/src/main/kotlin/org/wdt/wdtc/core/download/infterface/InstallTaskInterface.kt
package org.wdt.wdtc.core.download.infterface

import java.io.IOException

interface InstallTaskInterface {
  @Throws(IOException::class)
  fun overwriteVersionJson()

  @Throws(IOException::class)
  fun writeVersionJsonPatches()

  @Throws(IOException::class)
  fun afterDownloadTask()

  @Throws(IOException::class)
  fun beforInstallTask()
}
